v1.7.0/longhorn-engine/rockcraft.yaml

```diff
@@ -0,0 +1,267 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Rockcraft definition for Longhorn engine image:
+# longhornio/longhorn-engine:v1.7.0
+
+name: longhorn-engine
+summary: Rock containing Longhorn engine component.
+description: |
+  Rock containing Longhorn engine component: https://github.com/longhorn/longhorn-engine
+  Aims to replicate the upstream official image: longhornio/longhorn-engine:v1.7.0
+license: Apache-2.0
+
+version: "v1.7.0"
+
+# NOTE(aznashwan): the base for the engine image is the Suse Linux Enterprise
+# Base Container Image (SLE BCE) Service Pack 6 which ships with Linux 6.4,
+# and is thus most comparable to 24.04:
+# https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L1
+base: ubuntu@24.04
+
+build-base: ubuntu@24.04
+platforms:
+  amd64:
+  arm64:
+
+
+services:
+  longhorn-engine:
+    startup: enabled
+    override: replace
+
+    # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L68-L70
+    command: /tini -- [ longhorn ]
+
+parts:
+
+  # NOTE(aznashwan): the longhorn binary is built within a Docker container
+  # which is set up by Rancher's Dapper tool: https://github.com/rancher/dapper
+  # The setup steps for the build container are contained within this Dockerfile:
+  # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper
+  setup-build-env:
+    plugin: nil
+
+    build-packages:
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper#L30-L36
+      - cmake
+      - curl
+      - git
+      - less
+      - file
+      - gcc
+      - python3
+      - python3-pip
+      - python3-dev
+      - libkmod-dev
+      - libnl-3-dev
+      - linux-headers-6.8.0-41
+      - pkg-config
+      - psmisc
+      # - qemu-tools
```

Comment:
Do we need to replace this with `qemu-utils` too ?

Reply:
Great catch, no, I've just removed it!

The reason it was there to begin with is that the Longhorn folks use the Dapper-built Docker container image for both building and integration testing, so they basically install every runtime dependency in it, [including `qemu-img`](https://github.com/longhorn/longhorn-engine/blob/master/Dockerfile.dapper#L31).